CONTRIBUTING.md

```diff
@@ -1,5 +1,47 @@
 # How to contribute
 
-If you encounter any bugs with the library please file an issue in the [Issues](https://github.com/Azure/azure-sdk-for-java/issues) section of the project.
+Thank you for your interest in contributing to Azure SDK for Java
 
-If you would like to become an active contributor to this project please follow the instructions provided in [Microsoft Azure Projects Contribution Guidelines](http://azure.github.com/guidelines.html).
\ No newline at end of file
+# Ways to contribute
+- If you encounter any bugs with the library please file an issue in the [Issues](https://github.com/Azure/azure-sdk-for-java/issues) section of the project.
+
+
+## Getting started
+If you would like to become an active contributor to this project please follow the instructions provided in [Microsoft Azure Projects Contribution Guidelines](http://azure.github.com/guidelines.html).
+
+If you wish to make code changes, or contribute something new, please follow the [GitHub Forks / Pull requests model](https://help.github.com/articles/fork-a-repo/): Fork the repo, make the change and propose it back by submitting a pull request.
+
+
+## Setting up a developer environment
+- Install [Java 8](https://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
+  - add `JAVA_HOME` to environment variables ([Windows](https://javatutorial.net/set-java-home-windows-10), [Linux](https://docs.oracle.com/cd/E19062-01/sun.mgmt.ctr36/819-5418/gaznb/index.html), [Mac OS](http://www.sajeconsultants.com/how-to-set-java_home-on-mac-os-x/))
+- Install [Maven](http://maven.apache.org/download.cgi)
+  - add `MAVEN_HOME` to environment variables ([Windows](https://www.mkyong.com/maven/how-to-install-maven-in-windows/), [Linux](https://www.javahelps.com/2017/10/install-apache-maven-on-linux.html), [Mac OS](https://www.mkyong.com/maven/install-maven-on-mac-osx/))
+
+
+>**Note:** If you are on `Windows`, enable paths longer than 260 characters by: <br><br>
+Run this as Administrator on a command prompt:<br> 
+`REG ADD HKLM\SYSTEM\CurrentControlSet\Control\FileSystem /v LongPathsEnabled /t REG_DWORD /d 1`<br>(might need to type `yes` to override key if it already exists)<br><br>
+Set up with by running `git config --system core.longpaths true`
+
+## Building and Testing
+The easiest way to build is by running next command from root folder:
+> mvn -f pom.client.xml -Dgpg.skip -DskipTests clean install
```

Comment:
Use the code formatting syntax rather than the arrow - it formats poorly on GitHub otherwise.